SoC-2019-Microprojects.md

```diff
@@ -218,6 +218,28 @@ that we have `struct dir_iterator` (see dir-iterator.h), convert these
 to use the dir-iterator to simplify the code. Do only one conversion
 per microproject.
 
+### Let `git fetch <remote> HEAD` use the current branch name
+
+When fetching from a remote, one can specify which ref(s) to fetch.
+There is no convenient way to specify the current branch, though, and
+`HEAD` may be a good shortcut. Or `@`. Or `-`.
+
+### Teach `git range-diff` new tricks
+
+When contributing patches to Git, developers often need to compare
+in what form those patches made it upstream (e.g. to pick up fixes
+made by the Git maintainer in preparation for a new iteration). In
+such a case, a `--left-only` (or `--right-only`) option would come
+in real handy, to allow `git range-diff --left-only <branch>..upstream/pu`
+to skip commits that are upstream only (as there will be a metric
+ton of them).
+
+Another neat trick that could come in real handy at times would be
+to teach `git range-diff` to compare a local branch to a patch
+series that is provided as an mbox, e.g. to use `git range-diff` to see
+whether a given patch series on the mailing list has made it into
+git.git's `pu` branch yet.
```

Comment:
This idea seems a little big for a microproject to me as I don't think range-diff can process patches right now.

Reply:
Hey, this was a challenge for you, maybe you missed it... Just have a look at the implementation of `git range-diff`.

Here, I'll help you: https://github.com/git/git/blob/v2.21.0/range-diff.c#L27-L148

As you should have seen: `range-diff` simply takes the commit ranges, throws them at `git log -p` and then parses the output.

And *obviously* there is not a whole lot of work to do if you want to teach `range-diff` to read an mbox instead. We already have pretty much all the code that is needed, it just needs to be stitched together appropriately.

Reply:
I still feel that it is on the big side for a micro-project but I am ok to merge this anyway.

Reply:
If students are left entirely to their own devices, sure, it might be on the big side. But the point of GSoC is not cheap labor, is it? It is a student/mentor setup, and there is no good reason why that relationship should only start *after* the micro-project. Unless the mentor wants to avoid work, that is, in which case one might ask why they volunteered to mentor in the first place.